src/connectionController.ts

```diff
@@ -44,7 +44,7 @@ export interface StoreConnectionInfo {
   id: string; // Connection model id or a new uuid.
   name: string; // Possibly user given name, not unique.
   storageLocation: StorageLocation;
-  connectionOptions: ConnectionOptions;
+  connectionOptions?: ConnectionOptions;
```

Comment:
In old connection info object save to disc. We migrate those connections to a new format by removing connectionModel and adding connectionOptions.

Reply:
But in `_connections` the field is always there, correct? It might be nice to be explicit about that in the types and omit some of the type casts here, I left a suggestion about that below, feel free to check if it works